op-program/host/host.go

```diff
@@ -175,8 +175,14 @@ func PreimageServer(ctx context.Context, logger log.Logger, cfg *config.Config,
 	hinterDone = routeHints(logger, hintChannel, hinter)
 	select {
 	case err := <-serverDone:
+		if err := kv.Close(); err != nil {
+			return err
+		}
 		return err
 	case err := <-hinterDone:
+		if err := kv.Close(); err != nil {
+			return err
+		}
```

Comment:
I think we should close this in the `defer func()` on line 125 after the server and hinter are fully shutdown - otherwise there's a potential race condition where one of them is still access the database after we close it.